src/main/kotlin/com/atlassian/performance/tools/virtualusers/HttpClientWebDriver.kt

```diff
@@ -0,0 +1,334 @@
+package com.atlassian.performance.tools.virtualusers
+
+import com.google.common.util.concurrent.SettableFuture
+import net.jcip.annotations.GuardedBy
+import org.apache.http.auth.AuthScope
+import org.apache.http.auth.Credentials
+import org.apache.http.auth.UsernamePasswordCredentials
+import org.apache.http.client.CredentialsProvider
+import org.apache.http.impl.client.BasicCredentialsProvider
+import org.apache.http.impl.client.CloseableHttpClient
+import org.apache.http.impl.client.HttpClientBuilder
+import org.openqa.selenium.*
+import org.openqa.selenium.interactions.Keyboard
+import org.openqa.selenium.interactions.Mouse
+import org.openqa.selenium.remote.*
+import org.openqa.selenium.remote.internal.JsonToWebElementConverter
+import java.util.concurrent.Future
+import java.util.logging.Level
+
+internal class HttpClientWebDriver : RemoteWebDriver() {
```

Comment:
HTTP client pretending to be a WebDriver sounds like a good way to introduce a ton of mental overhead and bugs. We're promising abilities to take screenshots, take page source, access elements, etc. We fail most of these promises in runtime. This fake complexity spreads outwards, infecting the `Browser` and `VirtualUserBehavior` APIs.

Instead, we should generalize existing APIs. I didn't dig deep, but it seems the common structure is:
* actual traffic generator - WebDriver or HTTP client
** gotta be closeable
** injectable supplier - `Browser` SPI for WD, a `Supplier<CloseableHttpClient>` for HTTP 
* diagnostics - screenshot/page source/URL for WD, HTTP request/response dump for HTTP?

Reply:
It's an internal debt, but you're right, let's try to generalize the APIs instead of trying to fit into the existing one.